lib/Doctrine/Common/Proxy/AbstractProxyFactory.php

```diff
@@ -216,6 +226,13 @@ private function getProxyDefinition($className)
                     require $fileName;
                     break;
 
+                case self::AUTOGENERATE_FILE_OUTDATED:
+                    if ( ! file_exists($fileName) || filemtime($classMetadata->getReflectionClass()->getFileName()) > filemtime($fileName)) {
```

Comment:
I haven't know that proxy generation can be used for internal classes. In this case we shouldn't generate new proxy class. What do you think?